dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/TaskRemoteHostServiceImpl.java

```diff
@@ -0,0 +1,281 @@
+import org.apache.dolphinscheduler.plugin.task.api.model.SSHSessionHost;
+import org.apache.dolphinscheduler.plugin.task.api.ssh.DSSessionAbandonedConfig;
+import org.apache.dolphinscheduler.plugin.task.api.ssh.DSSessionPoolConfig;
+import org.apache.dolphinscheduler.plugin.task.api.ssh.SSHResponse;
+import org.apache.dolphinscheduler.plugin.task.api.ssh.SSHSessionHolder;
+import org.apache.dolphinscheduler.plugin.task.api.ssh.SSHSessionPool;
+import org.apache.dolphinscheduler.service.utils.Constants;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.BeanUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import com.baomidou.mybatisplus.core.metadata.IPage;
+import com.baomidou.mybatisplus.extension.plugins.pagination.Page;
+
+@Service
+public class TaskRemoteHostServiceImpl extends BaseServiceImpl implements TaskRemoteHostService {
+
+    private static final Logger logger = LoggerFactory.getLogger(TaskRemoteHostServiceImpl.class);
+
+    @Autowired
+    private TaskRemoteHostMapper taskRemoteHostMapper;
+
+    @Autowired
+    private TaskInstanceMapper taskInstanceMapper;
+
+    @Override
+    @Transactional
+    public int createTaskRemoteHost(User loginUser, TaskRemoteHostDTO taskRemoteHostDTO) {
+        checkTaskRemoteHostDTO(taskRemoteHostDTO);
+
+        checkOperatorPermissions(loginUser, null, AuthorizationType.TASK_REMOTE_TASK, TASK_REMOTE_HOST_CREATE);
+
+        if (isExistSameName(taskRemoteHostDTO.getName())) {
+            throw new ServiceException(Status.TASK_REMOTE_HOST_EXIST, taskRemoteHostDTO.getName());
+        }
+
+        TaskRemoteHost remoteHost = new TaskRemoteHost();
+        BeanUtils.copyProperties(taskRemoteHostDTO, remoteHost);
+        long remoteHostCode;
+        try {
+            remoteHostCode = CodeGenerateUtils.getInstance().genCode();
+        } catch (CodeGenerateException e) {
+            throw new ServiceException(Status.INTERNAL_SERVER_ERROR_ARGS);
+        }
+        remoteHost.setCode(remoteHostCode);
+        remoteHost.setOperator(loginUser.getId());
+        remoteHost.setCreateTime(new Date());
+        remoteHost.setUpdateTime(new Date());
+
+        int result = taskRemoteHostMapper.insert(remoteHost);
+        if (result > 0) {
+            permissionPostHandle(AuthorizationType.TASK_REMOTE_TASK, loginUser.getId(),
+                    Collections.singletonList(remoteHost.getId()), logger);
+            logger.info("Create remote host successes, host name {}.", remoteHost.getName());
+        }
```

Comment:
In this case, I think checking result > 0 is meanless, due to not multi db write operations after inserting. Defaultly, it will success if no exception. WDYT?
```suggestion
        taskRemoteHostMapper.insert(remoteHost);
        permissionPostHandle(AuthorizationType.TASK_REMOTE_TASK, loginUser.getId(),
                  Collections.singletonList(remoteHost.getId()), logger);
        logger.info("Create remote host successes, host name {}.", remoteHost.getName());
        
```

Reply:
done